lib/streamlit/type_util.py

```diff
@@ -725,7 +727,37 @@ def pyarrow_table_to_bytes(table: pa.Table) -> bytes:
     writer = pa.RecordBatchStreamWriter(sink, table.schema)
     writer.write_table(table)
```

Comment:
Just curious, since there is a recursive call, shall we move this below line 753 to prevent writing the table > 1 time?

Reply:
I think the current logic requires this in order to measure the byte length of the serialized table.  But we might be able to use the `table.nbytes` instead, which would not require the write. But it needs a bit more investigation how close the `nbytes` is to the serialized bytes.